services/src/main/java/org/keycloak/protocol/oidc/endpoints/ParEndpoint.java

```diff
@@ -0,0 +1,76 @@
+package org.keycloak.protocol.oidc.endpoints;
+
+import org.jboss.logging.Logger;
+import org.jboss.resteasy.annotations.cache.NoCache;
+import org.keycloak.authentication.ParResponse;
+import org.keycloak.events.EventBuilder;
+import org.keycloak.models.CodeToTokenStoreProvider;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.RealmModel;
+import org.keycloak.protocol.oidc.TokenManager;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.FormParam;
+import javax.ws.rs.POST;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+public class ParEndpoint {
+
+    private static final Logger LOG = Logger.getLogger(ParEndpoint.class);
+
+    public static final String OBJECT_ID = "objectId";
+    public static final String REDIRECT_ID = "redirectId";
+    public static final String OBJECT_TYPE = "objectType";
+    public static final String USERNAME = "username";
+    public static final String CLIENT_ID = "clientId";
+    public static final String REDIRECT_URI = "redirect_uri";
+
+    private final KeycloakSession session;
+    private final TokenManager tokenManager;
+    private final RealmModel realm;
+    private final EventBuilder event;
+
+    public ParEndpoint(KeycloakSession session, TokenManager tokenManager, RealmModel realm, EventBuilder event) {
+        this.session = session;
+        this.tokenManager = tokenManager;
+        this.realm = realm;
+        this.event = event;
+    }
+
+    @POST
+    @NoCache
+    @Consumes({MediaType.APPLICATION_FORM_URLENCODED})
+    public Response handlePar(@FormParam(OBJECT_ID) String objectId,
```

Comment:
Basically, this method has to receive and process parameters the same as ones in the current `AuthorizationEndpoint`. In addition, this method may receive some other parameters that are not defined by any standard documents like RFC 6749 and so on. [PAR's standard document](https://tools.ietf.org/html/draft-ietf-oauth-par-07#section-2.1) allows to support such extension (here, `object_id`, `object_type` and `username`). IMO, such extension have to be treated as Optional.

Reply:
I have implemented the map of parameters as the input data of this method. So, we can pass any key-value pairs.

Reply:
I've confirmed its change. Thank you.